db/models/model.js

```diff
@@ -53,9 +39,41 @@ function selectCommentsByArticleId(id) {
     });
 }
 
+function addComment(comment, id) {
+  return db
+    .query(
+      format(
+        `INSERT INTO comments (body, author, article_id) VALUES %L RETURNING *`,
+        [[comment.comment, comment.username, id]]
+      )
+    )
+    .then((result) => {
+      return result.rows;
+    })
+    .catch((err) => {
+      if (
+        err.detail === 'Key (author)=(abc) is not present in table "users".'
```

Comment:
Be cautious of this level of detail in your error handling, it won't work if someone used an invalid username that was anything other than 'abc' - instead consider checking the error code, like you have below.